source/framework/vx_image.c

```diff
@@ -376,6 +383,310 @@ static vx_status ownAllocImageBuffer(vx_reference ref)
+
+    for (i = 0; i < TIVX_SUBIMAGE_STACK_SIZE; ++i)
+    {
+        stack[i] = NULL;
+    }
+    stack[stack_pointer] = ref;
+    stack_pointer++;
+    while (0U < stack_pointer)
+    {
+        stack_pointer--;
+        local_img = stack[stack_pointer];
+        subimages = local_img->subimages;
+        tivx_obj_desc_image_t *obj_desc = (tivx_obj_desc_image_t *)local_img->base.obj_desc;
+        if (obj_desc->planes > 0U)
+        {
+            for (i = 0; i < obj_desc->planes; ++i)
+            {
+                obj_desc->mem_ptr[i].host_ptr = obj_desc->mem_ptr[i].host_ptr + offset[i];
+                obj_desc->mem_ptr[i].shared_ptr = tivxMemHost2SharedPtr(obj_desc->mem_ptr[i].host_ptr, (vx_enum)TIVX_MEM_EXTERNAL);
+            }
+        }
+        else
+        {
+            obj_desc->mem_ptr[0U].host_ptr = obj_desc->mem_ptr[0].host_ptr + offset[local_img->channel_plane];
+            obj_desc->mem_ptr[0U].shared_ptr = tivxMemHost2SharedPtr(obj_desc->mem_ptr[0U].host_ptr, (vx_enum)TIVX_MEM_EXTERNAL);
+        }
+        for (i = 0; i < TIVX_IMAGE_MAX_SUBIMAGES; ++i)
+        {
+            if (NULL != subimages[i])
+            {
+                if (TIVX_SUBIMAGE_STACK_SIZE < stack_pointer)
+                {
+                    VX_PRINT(VX_ZONE_ERROR, "Too many sub-images, may need to increase the value of TIVX_SUBIMAGE_STACK_SIZE\n");
+                    status = (vx_status)VX_ERROR_NO_RESOURCES;
+                    break;
+                }
+                else
+                {
+                    stack[stack_pointer] = subimages[i];
+                    stack_pointer++;
+                }
+            }
+        }
+    }
+    return status;
+}
+
+static vx_status swapImage(const vx_image input, const vx_image output)
+{
+    /* Swap image handles. Need to recalculate for all the sub-images
+        and sub-tensors.
+        NOTE that this will need updating if uniform images are
+        implemented in a more efficient manner!
+        lock only one reference as this is locking the global vx context
+    */
+    vx_status status = ownReferenceLock(&output->base);
+    if ((vx_status)VX_SUCCESS == status)
+    {
+        tivx_obj_desc_image_t *ip_obj_desc = (tivx_obj_desc_image_t *)input->base.obj_desc;
+        tivx_obj_desc_image_t *op_obj_desc = (tivx_obj_desc_image_t *)output->base.obj_desc;
+        vx_uint32 i;
+        vx_uint64 offsets[TIVX_IMAGE_MAX_PLANES];
+        vx_imagepatch_addressing_t addrs;
+        tivx_reference_callback_f destructor;
+        uint32_t creation_type;
+        uint32_t mem_size;
+        for (i = 0; i < TIVX_IMAGE_MAX_PLANES; ++i)
+        {
+            offsets[i] = op_obj_desc->mem_ptr[i].host_ptr - ip_obj_desc->mem_ptr[i].host_ptr;
+            addrs = op_obj_desc->imagepatch_addr[i];
+            op_obj_desc->imagepatch_addr[i] = ip_obj_desc->imagepatch_addr[i];
+            ip_obj_desc->imagepatch_addr[i] = addrs;
+            destructor = output->base.destructor_callback;
```

Comment:
@lcweaver Action: check to see if all images have the same destructor

Reply:
@raphacan I reviewed this and while these are generic callbacks, it is fine to leave the code as/is.  I'd suggest though adding a comment to say that this has been added for completeness and to ensure that in case there is ever a need for unique destructors, this has been accounted for

Reply:
comment added